core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -689,7 +689,7 @@ class ControllerApis(val requestChannel: RequestChannel,
   def handleCreatePartitions(request: RequestChannel.Request): Unit = {
     val future = createPartitions(request.body[CreatePartitionsRequest].data,
       authHelper.authorize(request.context, CREATE, CLUSTER, CLUSTER_NAME),
-      names => authHelper.filterByAuthorized(request.context, CREATE, TOPIC, names)(n => n))
+      names => authHelper.filterByAuthorized(request.context, ALTER, TOPIC, names)(n => n))
```

Comment:
We probably should pull out the CREATE -> ALTER fix since in theory you could be using KRaft + AclAuthorizer. If you want I can do that now (but it won't have a unit test, hope that's OK).

(It is tested by AuthorizerIntegrationTest but I wouldn't want to backport that to 3.{0,1})